Automattic/jetpack
# Verify Paid Newsletter Access level before checking User Subscriptions

## Background
There has been an overall increased connection to the DB and the hourly spike due to how we are gating comments. See `maybe_gate_existing_comments`. This `maybe_gate_existing_comments` filter is being called for all comments on WPCOM. To fix this, we first verify that a comment should be gated by checking the post access level. If the post should not be gated, we return early. 

The goal here is to not query the DB for every comment or post on WPCOM. We only query for posts/comments that could be potentially gated.

See p1684456374317729/1684412809.266309-slack-C02AVAR9B for more context. A quick fix was merged [here](https://github.com/Automattic/jetpack/pull/30355). This PR improves on that fix.


## Proposed changes:

* Verify Paid Newsletter access level should be gated before querying DB. 
* Replace newsletter references with post level. This can be confusing. See p1684457715342319/1684412809.266309-slack-C02AVAR9B
* Consolidate get_post_access_level functionality into the `Jetpack_Membership` class

### Other information:


## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






1. Create a post with a newsletter access level
2. Verify the post is gated correctly
3. Repeat steps 1 -2 with the another newsletter access level
4. Add the Subscribe block to a post
5. Verify the Subscribe functionality works as expected with a "free" and "paid" plan
